pkg/services/newrelic.go

```diff
@@ -109,6 +112,43 @@ type newrelicService struct {
 type newrelicDeploymentMarkerRequest struct {
 	Deployment NewrelicNotification `json:"deployment"`
 }
+type newrelicApplicationsResponse struct {
+	Applications []struct {
+		ID json.Number `json:"id"`
+	} `json:"applications"`
+}
+
+func (s newrelicService) getApplicationId(client *http.Client, appName string) (string, error) {
+	applicationsApi := fmt.Sprintf("%s/v2/applications.json?filter[name]=%s", s.opts.ApiURL, appName)
```

Comment:
I'd use the `url` library to construct this, to avoid `appName` potentially injecting arbitrary stuff